### DISCUSSION TITLE
Genesis 1.3 version 4 with the OPC propagation code

### DISCUSSION THREAD

**Discussion by sonafel** [svenreiche/Genesis-1.3-Version4#86] [Q&A] [answered]
Can Genesis 1.3 version 4, be combined with the OPC propagation code for simulating a FEL in a resonator configuration?

**Comment by svenreiche** [comment, marked as answer]
It should be easier now since Genesis is actually compiled into a static library and then linked against a very small executable. This would make it easier to call Genesis internally from a other program (such as OPC) or even python.

Alternative once can interface the two codes with the radiation dump field. The format has changed significantly from Version 2 to Version 3/4. I am not familiar with OPC to know which is its preferred format.


During the compilation the libraryis generated. One can install that to have a programmable interface with Genesis. but I would ask Peter van der Slot, which has done the interface between OPC and Genesis.

Best,
Sven

**Reply by dinhcnguyen** [reply]
Hello, I'm trying to download OPC for use with Genesis v.4. Could you please let me know where and how to download the correct version of OPC for Genensis v.4? Thanks.

**Reply by svenreiche** [reply]
I am actually not involved with the development of OPC. I would suggest to contact Peter van der Slot, who is the author of OPC.


Best
Sven
